lending_pool/src/utils.rs

```diff
@@ -0,0 +1,427 @@
+    #[view(getTotalCollateralAvailable)]
+    fn get_total_collateral_available(&self, account_position: u64) -> BigUint {
+        let mut deposited_amount_in_dollars = BigUint::zero();
+        let deposit_positions = self.deposit_position();
+        let deposit_position_iter = deposit_positions
+            .iter()
+            .filter(|dp| dp.owner_nonce == account_position);
+
+        for dp in deposit_position_iter {
+            let dp_data = self.get_token_price_data(dp.token_id);
+            deposited_amount_in_dollars += dp.amount * dp_data.price;
+        }
+
+        deposited_amount_in_dollars
+    }
+
+    #[view(getTotalBorrowedAmount)]
+    fn get_total_borrowed_amount(&self, account_position: u64) -> BigUint {
+        let mut total_borrowed_amount = BigUint::zero();
+        let borrow_positions = self.borrow_position();
+        let borrow_position_iter = borrow_positions
+            .iter()
+            .filter(|bp| bp.owner_nonce == account_position);
+
+        for bp in borrow_position_iter {
+            let bp_data = self.get_token_price_data(bp.token_id);
+            total_borrowed_amount += bp.amount * bp_data.price;
+        }
+
+        total_borrowed_amount
+    }
+
+    fn send_amount_in_dollars_to_liquidator(
+        &self,
+        initial_caller: ManagedAddress,
+        liquidatee_account_nonce: u64,
+        amount_to_return_to_liquidator_in_dollars: BigUint,
+    ) {
+        let mut amount_to_send = amount_to_return_to_liquidator_in_dollars;
+        let deposit_positions = self.deposit_position();
+        let deposit_position_iter = deposit_positions
+            .iter()
+            .filter(|dp| dp.owner_nonce == liquidatee_account_nonce);
+
+        // Send amount_to_return_in_dollars to initial_caller
+        for mut dp in deposit_position_iter {
+            let dp_data = self.get_token_price_data(dp.token_id.clone());
+            let amount_in_dollars_available_for_this_bp = &dp.amount * &dp_data.price;
+
+            if amount_in_dollars_available_for_this_bp <= amount_to_send {
+                // Send all tokens and remove DepositPosition
+                self.send()
+                    .direct_esdt(&initial_caller, &dp.token_id, 0, &dp.amount);
+                amount_to_send -= amount_in_dollars_available_for_this_bp;
+                self.deposit_position().swap_remove(&dp);
+
+                if amount_to_send == 0 {
+                    break;
+                }
+            } else {
+                // Send part of the tokens and update DepositPosition
+                let partial_amount_to_send = (&amount_to_send * BP / &dp_data.price) / BP;
+                self.send()
+                    .direct_esdt(&initial_caller, &dp.token_id, 0, &partial_amount_to_send);
+
+                // Update DepositPosition
+                self.deposit_position().swap_remove(&dp);
+                dp.amount -= partial_amount_to_send;
+                self.deposit_position().insert(dp);
```

Comment:
Shouldn't there be a `break` here? Since we've sent everything if we reached this branch.